### PR TITLE
Add session history

### PR DESCRIPTION
**Pull request by maranqz**
| Q             | A
| ------------- | ---
| New feature?  | yes
| Tests pass?   | yes, but not added test of new feature

[Screenshot pages](https://github.com/maranqz/yii2-usuario/tree/session-histroy/docs/helpful-guides/session-history)

### CODE REVIEW

**Comment by maxxer** [comment]
Hi. Thanks for the PR. I'll try to have a look soon

File: docs/helpful-guides/how-to-use-session-history.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
+How to enable session history
+============================
+
+Session history is list user sessions.
```
Comment (maxxer) [inline docs/helpful-guides/how-to-use-session-history.md]: I'd remove `is`

File: docs/helpful-guides/how-to-use-session-history.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+User can delete all session except current.
```
Comment (maxxer) [inline docs/helpful-guides/how-to-use-session-history.md]: all session**s**

File: docs/installation/configuration-options.md [hunk 3]
```diff
@@ -3,6 +3,22 @@ Configuration Options
...
+
+Number of expired storing records `session history`, values:
+- `false` Store all records without deleting
+- `integer` Count of records for storing
+
+#### timeoutSessionHistory (Type: `boolean, integer`, Default value: `false`)
+
+How long store `session history` after expiring, values:
+- `false` Store all records without deleting
+- `integer` Time for storing after expiring
```
Comment (maxxer) [inline docs/installation/configuration-options.md]: What unit `time` is? Minutes, days, hours..?

File: src/User/resources/i18n/ru/usuario.php [hunk 4]
```diff
@@ -302,4 +302,12 @@
     'Unable to disable two-factor authorization.' => '@@Не удалось отключить двухфакторную авторизацию.@@',
     'We couldn\'t re-send the mail to confirm your address. ' => '@@Мы не можем повторно отправить письмо для подтверждения вашего адреса электронной почты.@@',
     'We have sent confirmation links to both old and new email addresses. ' => '@@Мы отправили письма на ваш старый и новый почтовые ящики. Вы должны перейти по обеим, чтобы завершить процесс смены адреса.@@',
+    'User agent' => '',
```
Comment (maxxer) [inline src/User/resources/i18n/ru/usuario.php]: missing translation?

File: src/User/resources/views/admin/update.php [hunk 5]
```diff
@@ -67,6 +67,10 @@
                                                 'label' => Yii::t('usuario', 'Assignments'),
                                                 'url' => ['/user/admin/assignments', 'id' => $user->id],
                                             ],
+                                            [
+                                                'label' => Yii::t('usuario', 'Session history'),
+                                                'url' => ['/user/admin/session-history', 'id' => $user->id],
```
Comment (maxxer) [inline src/User/resources/views/admin/update.php]: This should be visible only if `enableSessionHistory` is true

File: src/User/resources/views/settings/_menu.php [hunk 6]
```diff
@@ -41,6 +41,10 @@
                 'items' => [
                     ['label' => Yii::t('usuario', 'Profile'), 'url' => ['/user/settings/profile']],
                     ['label' => Yii::t('usuario', 'Account'), 'url' => ['/user/settings/account']],
+                    [
+                        'label' => Yii::t('usuario', 'Session history'),
+                        'url' => ['/user/settings/session-history']
```
Comment (maxxer) [inline src/User/resources/views/settings/_menu.php]: This should be visible only if `enableSessionHistory` is true

File: src/User/Controller/AdminController.php [hunk 7]
```diff
@@ -346,4 +349,33 @@ public function actionForcePasswordChange($id)
         }
         $this->redirect(['index']);
     }
+
+    /**
+     * Display list session history
+     */
+    public function actionSessionHistory($id)
+    {
```
Comment (maxxer) [inline src/User/Controller/AdminController.php]: What about checking if the feature has been enabled?

File: src/User/Controller/AdminController.php [hunk 8]
```diff
@@ -346,4 +349,33 @@ public function actionForcePasswordChange($id)
...
+            'dataProvider' => $dataProvider,
+            'user' => $user,
+        ]);
+    }
+
+    /**
+     * Terminate all session user
+     */
+    public function actionTerminateSessions($id)
+    {
```
Comment (maxxer) [inline src/User/Controller/AdminController.php]: Again check if feature is enabled before allowing the action, maybe it should be done at `access` if possible

File: src/User/resources/i18n/ru/usuario.php (hunk 4 above)
Comment (maranqz) [reply src/User/resources/i18n/ru/usuario.php]: This phrase has not correct translation on russian and people use at it is. 

Should i replace '' on 'User agent'?

File: docs/helpful-guides/how-to-use-session-history.md (hunk 1 above)
Comment (maranqz) [reply docs/helpful-guides/how-to-use-session-history.md]: Fix it.

File: docs/helpful-guides/how-to-use-session-history.md (hunk 2 above)
Comment (maranqz) [reply docs/helpful-guides/how-to-use-session-history.md]: Fix it.

File: docs/installation/configuration-options.md (hunk 3 above)
Comment (maranqz) [reply docs/installation/configuration-options.md]: Fix it.

File: src/User/resources/views/admin/update.php (hunk 5 above)
Comment (maranqz) [reply src/User/resources/views/admin/update.php]: Fix it.

File: src/User/resources/views/settings/_menu.php (hunk 6 above)
Comment (maranqz) [reply src/User/resources/views/settings/_menu.php]: Fix it.

File: src/User/Controller/AdminController.php (hunk 7 above)
Comment (maranqz) [reply src/User/Controller/AdminController.php]: Fix it.

File: src/User/Controller/AdminController.php (hunk 8 above)
Comment (maranqz) [reply src/User/Controller/AdminController.php]: Fix it.

**Comment by maranqz** [comment]
Problem with travis.
symfony/event-dispatcher are installed v5 version (v4 was before) and codeception can not start correct on php 7.2 and 7.3.

[previous ](https://travis-ci.org/2amigos/yii2-usuario/jobs/612001723?utm_medium=notification&utm_source=github_status) build php 7.3 
[current](https://travis-ci.org/2amigos/yii2-usuario/jobs/618942563?utm_medium=notification&utm_source=github_status) build php 7.3 

Version of package can see opening "COMPOSER_MEMORY_LIMIT=-1 travis_retry composer install --prefer-dist --no-interaction"

File: src/User/resources/i18n/ru/usuario.php (hunk 4 above)
Comment (maxxer) [reply src/User/resources/i18n/ru/usuario.php]: I think it's better

**Comment by maxxer** [comment]
I don't see where the problem with `symfony/event-dispatcher` has been arised. I see no changes in `composer.*` files, and last commit pass tests

**Comment by maxxer** [comment]
Late, but thank you 👍

**Pull request merged**